pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            if should_restake_reward {
+                ledger.locked = ledger.locked.saturating_add(staker_reward);
+                Self::update_ledger(&staker, ledger);
+
+                // Update storage
+                GeneralEraInfo::<T>::mutate(current_era, |value| {
+                    if let Some(x) = value {
+                        x.staked = x.staked.saturating_add(staker_reward);
+                        x.locked = x.locked.saturating_add(staker_reward);
+                    }
+                });
+
+                CreatorEraStake::<T>::mutate(space_id, current_era, |staking_info| {
+                    if let Some(x) = staking_info {
+                        x.total = x.total.saturating_add(staker_reward);
+                    }
+                });
+
+                Self::deposit_event(Event::<T>::Staked {
+                    who: staker.clone(),
+                    creator: space_id,
+                    era: current_era,
+                    amount: staker_reward,
+                });
+            }
+
+            // TODO: mint tokens to balance before locking in case `should_restake_reward` is true
+            T::Currency::resolve_creating(&staker, reward_imbalance);
+            Self::update_staker_info(&staker, space_id, staker_info);
+
+            // TODO: change event
+            Self::deposit_event(Event::<T>::RewardsClaimed {
+                who: staker,
+                amount: staker_reward,
+            });
+
+            /*Ok(Some(if should_restake_reward {
+                T::WeightInfo::claim_staker_with_restake()
+            } else {
+                T::WeightInfo::claim_staker_without_restake()
+            })
+                .into())*/
+            Ok(().into())
+        }
+
+        #[pallet::call_index(8)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn claim_creator_reward(origin: OriginFor<T>, space_id: SpaceId, era: EraIndex) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let _ = ensure_signed(origin)?;
+
+            let creator_info =
+                RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+            let mut creator_stake_info =
+                Self::creator_stake_info(space_id, era).unwrap_or_default();
+
+            let creator_reward = Self::calculate_creator_reward(&creator_stake_info, &creator_info, era)?;
+
+            // Withdraw reward funds from the creators staking
+            let reward_imbalance = T::Currency::withdraw(
+                &Self::account_id(),
+                creator_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            T::Currency::resolve_creating(&creator_info.stakeholder, reward_imbalance);
+
+            // TODO: change event
+            Self::deposit_event(Event::<T>::RewardsClaimed {
```

Comment:
Need to be more specific:
```suggestion
            Self::deposit_event(Event::<T>::CreatorRewardsClaimed {
```